ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -86,12 +83,8 @@ internal constructor(
   private val activeSurvey: Survey = requireNotNull(surveyRepository.activeSurvey)
   private val job: Job =
     activeSurvey.getJob(requireNotNull(jobId)) ?: error("couldn't retrieve job for $jobId")
-  val tasks: List<Task> = buildList {
-    if (job.suggestLoiTaskType != null && loiId == null) {
-      add(createSuggestLoiTask(job.suggestLoiTaskType))
-    }
-    addAll(job.tasksSorted)
-  }
+  // LOI creation task is included only on "new data collection site" flow..
+  val tasks: List<Task> = job.tasksSorted.filter { loiId == null || !it.isAddLoiTask }
```

Comment:
I find this filter a little tricky to understand. Seems like we're getting tasks that either have no LOI Id or that are explicitly _not_ add LOI tasks. The latter makes sense given the comment, but I don't understand why we filter for tasks with `loiId == null`

Reply:
We're including all tasks when loiId = null. I can rewrite this, lmkwyt.